test/system/src/main/java/io/pravega/test/system/framework/kubernetes/K8sClient.java

```diff
@@ -204,12 +219,23 @@ public V1Namespace createNamespace(final String namespace) {
         K8AsyncCallback<V1PodList> callback = new K8AsyncCallback<>("listPods");
         api.listNamespacedPodAsync(namespace, PRETTY_PRINT, null, null, true, labelName + "=" + labelValue, null,
                                    null, null, false, callback);
-        return callback.getFuture()
-                       .thenApply(v1PodList -> {
-                           List<V1Pod> podList = v1PodList.getItems();
-                           log.debug("{} pod(s) found with label {}={}.", podList.size(), labelName, labelValue);
-                           return podList.stream().map(V1Pod::getStatus).collect(Collectors.toList());
-                       });
+        return callback.getFuture();
+    }
+
+    /**
+     * Method to fetch all restarted pods which match a label.
+     * @param namespace Namespace on which the pod(s) reside.
+     * @param labelName Name of the label.
+     * @param labelValue Value of the label.
+     * @return Future representing the list of pod status.
+     */
+    public CompletableFuture<Map<String, V1ContainerStatus>> getRestartedPods(String namespace, String labelName, String labelValue) {
```

Comment:
is it possible that this will cascade to other tests once first test fails with a restart?

Reply:
No, the code only fails a test if there is a restart of pods (with the provided label) during the course of execution of the test by comparing the restart count of the pods before and after the tests.